src/billing/billing-address-action-creator.ts

```diff
@@ -104,6 +114,33 @@ export default class BillingAddressActionCreator {
         });
     }
 
+    private _updateCustomerConsent(
+        {
+            email,
+            marketingEmailConsent,
+        }: GuestCredentials,
+        options?: RequestOptions
+    ): Observable<CustomerAction> {
+        if (!marketingEmailConsent) {
```

Comment:
there's no use case to require opting out. The opt out is most commonly offered via an unsubscribe link in the first email received. Same as we do we newsletter subscriptions in storefront.